tests/apitest_shared.py

```diff
@@ -0,0 +1,32 @@
+import os
+import pytest
+
+from personio_py import Personio, PersonioError
+
+# Personio client authentication
+CLIENT_ID = os.getenv('CLIENT_ID')
+CLIENT_SECRET = os.getenv('CLIENT_SECRET')
+personio = Personio(client_id=CLIENT_ID, client_secret=CLIENT_SECRET)
+
+shared_test_data = {}
+
+# deactivate all tests that rely on a specific personio instance
+try:
+    personio.authenticate()
+    can_authenticate = True
+    # This is used to ensure the test check for existing objects
+    test_employee = personio.get_employees()[0]
+    shared_test_data = {
+        'test_employee': {
+            'id': test_employee.id_,
+            'first_name': test_employee.first_name,
+            'last_name': test_employee.last_name,
+            'email': test_employee.email,
+            'hire_date': test_employee.hire_date
+        }
+    }
```

Comment:
I don't think it's a good idea to have this shared state here, it's too easy to mess things up when one test writes to this dictionary while another expects the dictionary to be in a different state.
There can be reasons to share state between tests (e.g. the personio client which holds authentication parameters), but other than that I would avoid it.

Reply:
This is the only place where data is added to this dict.
Its purpose is to have a user object available for tests which is also available on the personio servers.

Reply:
this may be the only place where you are adding data to this dict *right now*, but it is not obvious to anyone else, what this dictionary is even good for. Also, this means that a request is made to the personio server *every time* this python file is imported, whether you are executing a test or not.

Please move this to a function, if you really need it for your test cases. You can use [memoization](https://docs.python.org/3/library/functools.html#functools.cache) if you want to avoid the HTTP request on every function call.

Reply:
You need some kind of user retrieval for the online test cases - how to otherwise make sure the user exists on the server instance? Currently all your online test fail on my test instance - that's why I retrieve a user first.

Reply:
Implemented using lru_cache for retrieval function. Shared dict is removed